core/sts-3.3.md

```diff
@@ -0,0 +1,94 @@
+---
+title: IRCv3.3 Strict Transport Security
+layout: spec
+copyrights:
+  -
+    name: "Attila Molnar"
+    period: "2015"
+    email: "attilamolnar@hush.com"
+---
+
+## Description
+
+Strict Transport Security (STS) is a mechanism which allows servers to
+communicate that complying clients should only connect to them over a secure
+connection.
+
+The policy is communicated to the client via the `sts` capability and should
+be processed by the client at capability negotiation time.
+
+The value of the capability specifies the duration during which the client
+MUST only connect to the server securely (using TLS, aka. SSL), and the port
+number to use for it.
+
+## Details
+
+The capability value, which MUST always be present, is a comma (`,`) separated
+list of tokens. Each token is a key which may have a value attached. If there
+is a value attached, the value is separated from the key by a `=`. That is,
+`<key>[=<value>][<key2>[=<value2>][<keyN>[=<valueN>]]]`. Keys specified in
+this document MUST only occur at most once.
+
+See [capability negotiation 3.2](capability-negotiation-3.2.html) for more
+information about capabilities with values.
+
+When the client sees the `sts` capability advertised over a non-secure
+connection, it MUST first establish a secure connection and confirm that the
+`sts` capability is still present.
+
+Once the client has confirmed that the `sts` capability is indeed offered over
+a secure connection, it then MUST only attempt secure connections to the
+server from now on. It MUST refuse to connect if a secure connection cannot
+be established with the server for any reason.
+
+If the client cannot establish a secure connection for any reason, it MAY
+continue using the insecure connection.
+
+Clients SHOULD NOT request this capability (`CAP REQ`).
```

Comment:
If client sees this in 005, client can upgrade to TLS using STARTTLS immediately, without reconnect.
But on the next connect, client will remember (using sts) that server supported tls, and will upgrade connection (or use another port) before registration